ui/app/src/components/NewContentDialog/NewContentDialogContainer.tsx

```diff
@@ -96,16 +99,21 @@ export function NewContentDialogContainer(props: NewContentDialogContainerProps)
           ? `${selectedItem.path}/`
           : selectedItem.path;
 
-      fetchLegacyContentTypes(site, path).subscribe(
-        (response) => {
-          setContentTypes(response);
+      fetchLegacyContentTypes(site, path).subscribe({
+        next(response) {
+          if (response.length === 1) {
+            dispatch(closeNewContentDialog());
```

Comment:
Who closes the dialog when there are multiple content types?
I feel closing the dialog should be done by the consumer of the dialog.

Reply:
The original behavior (and now the behavior when there are more than 1 content-type available) is that it closes after the form has been saved, so it stays open while a form in open (in case it's cancelled you can select another content-type). That behavior is done in this component though.